flowcraft/templates/process_tsv.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+"""
+Purpose
+-------
+This module is intended to process the output in tsv
+ to generate a report in json format.
+Expected input
+--------------
+The following variables are expected whether using NextFlow or the
+:py:func:`main` executor.
+- ``sample_id`` : Sample Identification string.
+- ``tsv``: tsv output.
+Generated output
+----------------
+- ``.report.jason``: Data structure for the report
+Code documentation
+------------------
+"""
+
+import json
+import csv
+import os
+
+from flowcraft_utils.flowcraft_base import get_logger, MainWrapper
+
+__version__ = "1.0.1"
+__build__ = "05.10.2018"
+__template__ = "maxbin2-nf"
+
+logger = get_logger(__file__)
+
+if __file__.endswith(".command.sh"):
+    SAMPLE_ID = '$sample_id'
+    FILE = '$tsv'
+    logger.debug("Running {} with parameters:".format(
+        os.path.basename(__file__)))
+    logger.debug("SAMPLE_ID: {}".format(SAMPLE_ID))
+    logger.debug("FILE: {}".format(FILE))
+
+@MainWrapper
+def main(sample_id, tsv_file):
+
+    with open(".report.json", "w") as k:
+        k.write('{"tsvData":[{"sample":"' + sample_id + '", "data": { "MaxBin2":')
+        k.write(json.dumps(list(csv.reader(open(tsv_file), delimiter='\t'))))
+        k.write('}}]}')
```

Comment:
I am not sure of what is going on here, but this seems a rather poor writing to the file :smirk:. Maybe you should load the content of `tsv_file` into a dict and then use `json.dumps` to dump the whole dict instead of this sketchy solution.
Basically define a dict like this:

```python
# first define a backbone of your dict like this
your_dict = {
    "tsvData": [{
        "sample": sample_id,
        "data": {}
    }]
}
# then add your entry to the dict. I am not sure why you use a list for `tsvData` since you put everything into `MaxBin2` but I hope you get the idea anyway.
your_dict["tsvData"][0]["data"]["MaxBin2"] = list(csv.reader(open(tsv_file)
# then json dump it to file like this
k.write(json.dumps(your_dict))
``` 

The whole idea is to construct a dict before dumping into the file rather than sketching a dict to a file with strings